src/modals/masq_favorite_modal.js

```diff
@@ -0,0 +1,56 @@
+import MasqFavoriteModalView from '../views/masq_favorite_modal.dot'
+import Modal from "./modal"
+import Store from "../adapters/store"
+import {version} from '../../config/constants.yml'
+
+const store = new Store()
+
+export default class MasqFavoriteModal {
+  constructor() {
+    // get masqFavoriteModal from window if already initialized
+    if (window.__masqFavoriteModal) {
+      return window.__masqFavoriteModal
+    }
+    // if masqFavoriteModal not initialized, use this
+    window.__masqFavoriteModal = this
+
+    this.modal = new Modal(this, MasqFavoriteModalView)
+
+    this.eventTarget = document.createElement('masqFavoriteModal')
+
+    this.opened = false
+  }
+
+
+  open () {
+    let doNotAskAgain = localStorage.getItem(`qmaps_v${version}_add_favorite_ask_masq`)
+    if (doNotAskAgain === "false" || doNotAskAgain === null) {
+      this.opened = true
+      this.modal.open()
+    }
+  }
+
+  close () {
+    const doNotAskAgain = document.getElementById("doNotAsk").checked
+    localStorage.setItem(`qmaps_v${version}_add_favorite_ask_masq`, doNotAskAgain)
+    this.modal.close()
+    this.eventTarget.dispatchEvent(new Event('closed'))
+    this.opened = false
+  }
+
+  async openMasq () {
+    await store.login()
+    if (store.isLoggedIn()) {
+      this.close()
+    }
+  }
+
+  waitForClose() {
+    if (!this.opened) {
```

Comment:
The masq_favorite_modal is opened when a user that is not connected to masq tries to add a favorite to offer him to connect to masq. In the case he connects to masq through the modal, we have to add the favorite in his masq profile, otherwise we add it to the local storage as usual. That's why we need to wait for the modal to close before doing something in the poi panel, that's how the waitForClose promise is used.

There are alternative such as using events or just calling store.add in the modal component before closing so the poi_panel doesn't have to await the closing of the modal.